app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -820,34 +820,30 @@ private void updateStatusText() {
         }
     }
 
-    public static boolean copyFile(InputStream inputStream, OutputStream out) {
+    public static void copyFile(InputStream inputStream, OutputStream out) throws IOException {
         Logger.log(TAG, "File server copying file");
-        Log.d(CommCareWiFiDirectActivity.TAG, "Copying file");
         if (inputStream == null) {
-            Log.d(CommCareWiFiDirectActivity.TAG, "Input Null");
+            Logger.log(TAG, "Input stream null");
+            throw new IOException("Got null input stream");
         }
         byte buf[] = new byte[1024];
         int len;
         try {
             while ((len = inputStream.read(buf)) != -1) {
-                Log.d(CommCareWiFiDirectActivity.TAG, "Copying file : " + new String(buf));
                 out.write(buf, 0, len);
             }
             out.close();
             inputStream.close();
         } catch (IOException e) {
-            Log.d(CommCareWiFiDirectActivity.TAG, e.toString());
-            Logger.log(TAG, "Copy in File Server failed");
-            return false;
+            Logger.log(TAG, "Copy in File Server failed with exception " + e);

```

Comment:
Is this exception going to be captured somewhere where we'll log the stack trace?